runtime/lua/vim/shared.lua

```diff
@@ -267,18 +267,23 @@ function vim.tbl_deep_extend(behavior, ...)
 end
 
 --- Deep compare values for equality
+---
+--- Tables are compared recursively unless they both provide the `eq` methamethod.
+--- All other types are compared using the equality `==` operator.
+--@param a first value
+--@param b second value
+--@returns `true` if values are equals, else `false`.
```

Comment:
format was updated, see `:help dev-lua-doc`

```suggestion
---@param a first value
---@param b second value
---@returns `true` if values are equals, else `false`.
```